app/src/main/java/org/eyeseetea/malariacare/layout/score/ScoreRegister.java

```diff
@@ -206,14 +206,17 @@ public static Float calcNum(Question question, float idSurvey) {
             return null;
         }
         Value value = question.getValueBySurvey(idSurvey);
-        //Returns null if the question will be ignored(not compulsory, and not answered or child with inactive parent questions)
+
+        //when a child question is compulsory but is not active, its ignored
+        if(question.getCompulsory() && question.hasParent()){
```

Comment:
@idelcano what is this? Three 'if' one inside the other and finally a 'return null'. Can this code be improved? Same for the 'if' below